packages/mobile/src/components/ContactCircle.tsx

```diff
@@ -0,0 +1,112 @@
+import colors from '@celo/react-components/styles/colors'
+import fontStyles from '@celo/react-components/styles/fonts'
+import { getContactNameHash } from '@celo/utils/src/contacts'
+import * as React from 'react'
+import { Image, StyleSheet, Text, View, ViewStyle } from 'react-native'
+import { MinimalContact } from 'react-native-contacts'
+import { useSelector } from 'react-redux'
+import DefaultAvatar from 'src/icons/DefaultAvatar'
+import { addressToDisplayNameSelector } from 'src/identity/reducer'
+
+interface Props {
+  style?: ViewStyle
+  contact?: MinimalContact
+  name: string | null // Requiring a value so we need to be explicit if we dont have it
+  address?: string
+  size?: number
+  thumbnailPath?: string | null
+}
+
+const DEFAULT_ICON_SIZE = 40
+export const contactIconColors = [colors.teal, colors.orange, colors.purple]
+
+const getAddressBackgroundColor = (address: string) =>
+  `hsl(${parseInt(address.substring(0, 5), 16) % 360}, 53%, 93%)`
+const getAddressForegroundColor = (address: string) =>
+  `hsl(${parseInt(address.substring(0, 5), 16) % 360}, 67%, 24%)`
+const getContactColor = (contact: MinimalContact) =>
+  getAddressBackgroundColor(getContactNameHash(contact))
+const getContactInitial = (contact: MinimalContact) => getNameInitial(contact.displayName)
+const getNameInitial = (name: string) => name.charAt(0).toLocaleUpperCase()
+
+function ContactCircle({ contact, size, thumbnailPath, name, address, style }: Props) {
+  const addressToDisplayName = useSelector(addressToDisplayNameSelector)
+  const addressInfo = addressToDisplayName[address || '']
+  const displayName = name || addressInfo?.name
+  const iconSize = size || DEFAULT_ICON_SIZE
+  const iconBackgroundColor =
+    (contact && getContactColor(contact)) ||
+    (address && getAddressBackgroundColor(address)) ||
+    contactIconColors[0]
+
+  const getInitials = () =>
+    (contact && getContactInitial(contact)) || (displayName && getNameInitial(displayName)) || '#'
+
+  const renderThumbnail = () => {
+    const resolvedThumbnail =
+      thumbnailPath || (contact && contact.thumbnailPath) || addressInfo?.imageUrl
+    const iconSize = size || DEFAULT_ICON_SIZE
+
+    if (resolvedThumbnail) {
+      return (
+        <Image
+          source={{ uri: resolvedThumbnail }}
+          style={[
+            styles.image,
+            { height: iconSize, width: iconSize, borderRadius: iconSize / 2.0 },
```

Comment:
If we can use `iconSize` defined outside of this function, we should be able to compute this style in advance, so it would not be recreated on each re-render.

Reply:
We can use the `iconSize` but how would we compute in advance? The only thing I can think of is doing that inside a `useMemo` but I fear that might negatively affect readability and the performance impact of recalculating this seems very low, especially because the props of this class don't change very often if at all so it won't re-render many times anyways.